File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.cpp
using namespace std;

class Solution {
public:
    int numberOfBeams(vector<string>& bank) 
    {
        vector<int> devices;    

        for (int i = 0; i < bank.size(); i++)
        {
            int count = 0;
            for (char c : bank[i])
                if (c == '1')
                    count++;
            if (count>0)
                devices.push_back(count);
        }

        int beams = 0;
        if(devices.empty())
            return beams;
        for (int i = 0; i < devices.size()-1; i++)
            beams += devices[i]*(devices[i+1]);
        return beams;
    }
};

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        devices = [bankrow.count("1") for bankrow in bank]
        beams = 0

        for i in range(len(devices)):
            if devices[i] != 0:
                for j in range(i+1, len(devices)):
                    if devices[j] != 0:
                        beams += devices[i] * devices[j]
                        break
        return beams